plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py

```diff
@@ -64,11 +65,25 @@ def run(self, params={}, state={}, custom_config={}):  # pylint: disable=unused-
             # Force 'now' to be 15 minutes before now as CB Analytics alerts can be updated for up to 15 minutes
             # following the original backend_timestamp, after which time the alert is considered immutable.
             now = now_time - timedelta(minutes=15)
-            end_time = now.strftime(TIME_FORMAT)
 
             # Check if we have made use of custom config to change the start times from DEFAULT_LOOKBACK
-            alerts_start, observations_start, alert_page_size, observation_page_size, debug = self._parse_custom_config(
-                custom_config, now, state
+            (
+                alerts_start,
+                observations_start,
+                alert_page_size,
+                observation_page_size,
+                observation_window,
+                debug,
+            ) = self._parse_custom_config(custom_config, now, state)
+
+            # calculate end window
+            calculated_end_window = datetime.strptime(observations_start, TIME_FORMAT).astimezone(
```

Comment:
we might need to only calculate this based on the alert start. maybe better to have defined alert_end and observation_end? usually alerts we can keep up ok, so this would move it forward when we don't want to